tests/testthat/test-carehomes-multistrain.R

```diff
@@ -1476,32 +1850,148 @@ test_that("G_D empty when p_G_D = 0", {
                             strain_seed_rate = c(10, 0),
                             strain_seed_date =
-                              sircovid_date(c("2020-02-07", "2020-02-08")))
+                              sircovid_date(c("2020-02-07", "2020-02-08")),
+                            model_super_infection = 1)
 
   mod <- carehomes$new(p, 0, np, seed = 1L)
 
   initial <- carehomes_initial(mod$info(), np, p)
   mod$set_state(initial$state, initial$step)
-  index_G_D <- mod$info()$index$G_D
-  index_G_D_strain_1 <- index_G_D[c(1:19, 39:57)]
-  index_G_D_strain_2 <- index_G_D[c(20:38, 58:76)]
 
   end <- sircovid_date("2020-05-01") / p$dt
   steps <- seq(initial$step, end, by = 1 / p$dt)
   set.seed(1)
-  y <- mod$simulate(steps)
+  y <- mod$transform_variables(
+    drop(mod$simulate(steps)))
+
+  # Strain 1 and 4 (2 -> 1)
+  expect_false(all(y$G_D[, 1, , , , ] == 0))
+  expect_false(all(y$G_D[, 4, , , , ] == 0))
+  # Strain 2 and 3 (1 -> 2)
+  expect_true(all(y$G_D[, 2, , , , ] == 0))
+  expect_true(all(y$G_D[, 3, , , , ] == 0))
+})
+
+
+test_that("Can't move from S to E3/4", {
+  np <- 3L
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            strain_transmission = c(1, 1),
+                            strain_seed_rate = c(10, 0),
+                            strain_seed_date =
+                              sircovid_date(c("2020-02-07", "2020-02-08")),
+                            model_super_infection = 1)
+
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+
+  initial <- carehomes_initial(mod$info(), np, p)
+  mod$set_state(initial$state, initial$step)
+  end <- sircovid_date("2020-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+  set.seed(1)
+  y <- mod$transform_variables(
+    drop(mod$simulate(steps)))
+  expect_true(any(y$E[, 1, , , , 2] > 0))
+  expect_true(any(y$E[, 2, , , , 2] > 0))
+  expect_true(all(y$E[, 3:4, , , , 2] == 0))
+})
+
+
+test_that("Everyone in R1 when strain_transmission = c(1, 0)", {
+  np <- 3L
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            strain_transmission = c(1, 0),
+                            strain_seed_rate = c(10, 0),
+                            beta_value = 1e100,
+                            strain_seed_date =
+                              sircovid_date(c("2020-02-07", "2020-02-08")),
+                            model_super_infection = 1)
+
+  mod <- carehomes$new(p, 0, np, seed = 1L)
+
+  initial <- carehomes_initial(mod$info(), np, p)
+  mod$set_state(initial$state, initial$step)
+  end <- sircovid_date("2020-05-01") / p$dt
+  steps <- seq(initial$step, end, by = 1 / p$dt)
+  set.seed(1)
+  y <- mod$transform_variables(
+    drop(mod$simulate(steps)))
+
+  expect_true(all(y$S[, , , 85] == 0))
+  expect_true(all(y$R[, 2:4, , , 85] == 0))
+  expect_false(all(y$R[, 1, , , 85] == 0))
+})
+
+
+test_that("Can only move from R3 and R4 to S", {
```

Comment:
I would rename the test "Can only move to S from R3 and R4" ( you can move to S from R1 and R2)